financial-connections/src/main/java/com/stripe/android/financialconnections/domain/FlowCoordinator.kt

```diff
@@ -0,0 +1,82 @@
+package com.stripe.android.financialconnections.domain
+
+import com.stripe.android.core.Logger
+import com.stripe.android.financialconnections.domain.FlowCoordinatorMessage.RequestNextStep
+import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest
+import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest.FinancialConnectionsAuthorizationSession
+import com.stripe.android.financialconnections.navigation.NavigationCommand
+import kotlinx.coroutines.flow.MutableSharedFlow
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * The AuthFlow state is centralized in the parent viewModel.
+ *
+ * This component acts as a communication channel used by steps to send messages to parent
+ * [com.stripe.android.financialconnections.presentation.FinancialConnectionsSheetNativeViewModel]
+ *
+ */
+@Singleton
+internal class FlowCoordinator @Inject constructor() {
```

Comment:
It seems this is a singleton bound to FinancialConnectionsSheetNativeViewModel, do you think it makes more sense to rename more specificly? (e.g `NativeFlowCoordinator`)? Otherwise I would think this is singleton related to the entire ConnectionSheet

Reply:
Renamed to `NativeAuthFlowCoordinator` - https://github.com/stripe/stripe-android/pull/5140/commits/e3f2e13001ecbeb1b6df33cd3225e8d8d10df5a6